backend/logic/users/users.js

```diff
@@ -1,112 +1,48 @@
+const mysql = require('../../sql/connection');
+const format = require('../../validation/format');
+const hasher = require('../../validation/hash');
 const util = require("util");
 
-const mysql = require("../../sql/connection");
-const format = require("../../validation/format");
-const hasher = require("../../validation/hash");
-
-var insertStudentUser = (username, password, email, id) => {
-  return new Promise(function(res, rej) {
-    // Connect to database
-    let promise = new Promise(function(resolve, reject) {
-      mysql.getConnection(function(err, conn) {
-        if (err) {
-          reject(err);
-        }
-        resolve(conn);
-      });
-    });
-
-    // Await for connection
-    promise.then(
-      function(value) {
-        let connection = value;
-        let error;
-
-        // Check for missing input params
-        if (!username) {
-          error = "undefined username";
-        } else if (!password) {
-          error = "undefined password";
-        } else if (!email) {
-          error = "undefined email";
-        } else if (!id) {
-          error = "undefined id";
-        }
-
-        if (error) {
-          connection.release();
-          console.error(error);
-          res(error);
-          return;
-        }
-
-        // Check for invalid formatting
-        if (!format.verifyUsername(username)) {
-          error = "invalid format username";
-        } else if (!format.verifyPassword(password)) {
-          error = "invalid format password";
-        } else if (!format.verifyEmail(email)) {
-          error = "invalid format email";
-        } else if (!format.verifyId(id)) {
-          error = "invalid format id";
-        }
-
-        if (error) {
-          connection.release();
-          console.error(error);
-          res(error);
-          return;
-        }
+var insertStudentUser = async (username, password, email, id) => {
+  // Connect to database
+  let connection = await mysql.getNewConnection()
```

Comment:
getNewConnection() can be moved to the try catch block

Reply:
thats true thanks!